src/base.rs

```diff
@@ -0,0 +1,290 @@
+//! Primitives for object and array transmutation.
+//!
+//! The functions in this module are very unsafe and their use is not
+//! recommended unless you _really_ know what you are doing.
+use crate::error::Error;
+use crate::guard::{Guard, SingleManyGuard, SingleValueGuard, PermissiveGuard};
+#[cfg(feature = "std")]
+use core::mem::forget;
+use core::mem::size_of;
+use core::slice;
+
+/// Convert a byte slice into a single instance of a `Copy`able type.
+///
+/// The byte slice must have at least enough bytes to fill a single instance of
+/// a type, extraneous data is ignored.
+///
+/// # Safety
+///
+/// - This function does not perform memory alignment checks. The beginning of
+/// the slice data must be properly aligned for accessing the value of type `T`.
+/// - The byte data needs to correspond to a valid `T` value.
+///
+/// Failure to fulfill any of the requirements above is dangerous and may eat
+/// your laundry.
+///
+/// # Errors
+///
+/// An error is raised if the slice does not have enough bytes for a single
```

Comment:
"returned" – in keeping with stdlib docs panics are "raised", but errors are "returned".

Occurs on:
  *	L28
  *	L70
  *	L117